src/components/CustomTileModal/index.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useState } from 'react'
+
+import { Modal } from '@entur/modal'
+import { PrimaryButton, SecondaryButton } from '@entur/button'
+import { Radio, RadioGroup, TextArea, TextField } from '@entur/form'
+
+import { useSettingsContext } from '../../settings'
+
+import './styles.scss'
+
+interface Props {
+    setIsOpen: (isOpen: boolean) => void
+    selectedTileId?: string
+}
+
+type TileType = 'qr' | 'image'
+
+type ActionType = 'update' | 'addNew'
+
+const CustomTileModal = ({ setIsOpen, selectedTileId }: Props): JSX.Element => {
+    const [settings, setSettings] = useSettingsContext()
+    const { customQrTiles = [], customImageTiles = [] } = settings || {}
+    const selectedItem = [...customQrTiles, ...customImageTiles].find(
+        (tile) => tile.id === selectedTileId,
+    )
+    const [tileType, setTileType] = useState<TileType>(
+        selectedItem && 'displayHeader' in selectedItem ? 'image' : 'qr',
+    )
+    const [displayName, setDisplayName] = useState(
+        selectedItem ? selectedItem.displayName : '',
+    )
+    const [linkAddress, setLinkAddress] = useState(
+        selectedItem ? selectedItem.linkAddress : '',
+    )
+    const [description, setDescription] = useState(
+        selectedItem ? selectedItem.description : '',
+    )
+    const [displayHeader, setDisplayHeader] = useState(
+        selectedItem && 'displayHeader' in selectedItem
+            ? selectedItem['displayHeader']
+            : '',
+    )
+
+    const [isSubmitAttempted, setIsSubmitAttempted] = useState(false)
+
+    const handleSubmit = (actionType: ActionType) => {
+        setIsSubmitAttempted(true)
+        if (!displayName || !linkAddress) return
+        if (tileType === 'qr') {
```

Comment:
Kan qr og image være en enum i stedet for string? :)

Reply:
kanskje det samme med actionType også?

Reply:
Ja, det kan gjøres om, men er det egentlig bedre? Har lest meg litt opp på det og ser at det er litt debatt. Personlig foretrekker jeg union types. Legger ved to kilder her:

https://stackoverflow.com/questions/40275832/typescript-has-unions-so-are-enums-redundant
https://blog.bam.tech/developer-news/should-you-use-enums-or-union-types-in-typescript

Reply:
Bump @caroline-odden :)

Reply:
Tja, personlig liker jeg enums, for det utelukker "skrivefeil" når man er en del utviklere. Brått innfører noen 'QR' i stedet for 'qr' for forvirringen øker.

Reply:
Ja, hadde det kun vært "vanlige" strenger hadde jeg vært enig. Men når man gjør det som union type ser jeg ikke hvordan problemet med skrivefeil vil kunne oppstå. Se f. eks. følgende eksempel hvor man skal velge actionType. Til å begynne med får man (seff avhenggig av editor og innstillinger etc) følgende forslag: 

![image](https://user-images.githubusercontent.com/32391231/144866914-634f71a1-1eaf-48ca-bf04-69424424d476.png)

Skriver man feil, f. eks. feil casing, får man beskjed:

![image](https://user-images.githubusercontent.com/32391231/144867088-d12eb402-5974-43f5-b12b-80713ca96f74.png)

Og til sist så må man spesifikt surpresse (mulig man kan omgå den i tsconfig) om man vil få godkjent `tsc` som er del av pipelinen vår på circleci

![image](https://user-images.githubusercontent.com/32391231/144867636-27e478e8-6e4d-4121-8e85-f6dadae2f9c1.png)

Reply:
Sant, det kan godt være en type :)